manifests/init.pp

```diff
@@ -536,6 +538,20 @@
     }
   }
 
+  case type3x($ssh_key_export) {
+    'string': {
+      validate_re($ssh_key_export, '^(true|false)$', "ssh::ssh_key_export may be either 'true' or 'false' and is set to <${ssh_key_export}>.")
+      $ssh_key_export_real = str2bool($ssh_key_export)
+    }
+    'boolean': {
+      $ssh_key_export_real = $ssh_key_export
+    }
+    default: {
+      fail('ssh::ssh_key_export type must be true or false.')
```

Comment:
I literally copy + pasted the ssh_key_import code and changed the variable names - if you want it written better, that's not a big deal, I'll do it. Just realize there's more of this in there.